solr/test-framework/src/java/org/apache/solr/SolrTestCase.java

```diff
@@ -144,12 +152,107 @@ public static void checkSyspropForceBeforeClassAssumptionFailure() {
   public void checkSyspropForceBeforeAssumptionFailure() {
     // ant test -Dargs="-Dtests.force.assumption.failure.before=true"
     final String PROP = "tests.force.assumption.failure.before";
-    assumeFalse(PROP + " == true",
-                systemPropertyAsBoolean(PROP, false));
+    assumeFalse(PROP + " == true", systemPropertyAsBoolean(PROP, false));
   }
-  
+
   @AfterClass
-  public static void shutdownLogger() throws Exception {
+  public static void shutdownLogger() {
     StartupLoggingUtils.shutdown();
   }
+
+  protected static void interruptThreadsOnTearDown() {
```

Comment:
This can be done (or maybe is, didn't look at the current annotation on SolrTestCase/LuceneTestCase) automatically by the test framework -- see thread lingering/ actions on these annotations.

https://github.com/randomizedtesting/randomizedtesting/blob/master/examples/maven/src/main/java/com/carrotsearch/examples/randomizedrunner/Test010Lingering.java#L40-L43